lib/statsd/instrument/expectation.rb

```diff
@@ -36,7 +36,7 @@ def initialize(client: StatsD.singleton_client, type:, name:, value: nil,
         sample_rate: nil, tags: nil, no_prefix: false, times: 1)
 
         @type = type
-        @name = client.prefix ? "#{client.prefix}.#{name}" : name unless no_prefix
+        @name = client.prefix && !no_prefix ? "#{client.prefix}.#{name}" : name
```

Comment:
I couldn't comment in time because of the GH outage, but I find this condition hard to grasp.

`no_prefix || !client.prefix` would have been much more readable.

Reply:
Fixed on master